### PR TITLE
Use get_data internally instead of get_data_from_viewer

### PR DESCRIPTION
**Pull request by rosteen**
Note that I left the call in `app.get_subsets_from_viewer`, since I think that will be part of @javerbukh's subset work, although I did implement something rather related in `specviz.get_spectra`. 

I believe there is one last failing test, and I probably need to get rid of some references to the old method in the docs. Other than that I think this is ready for other eyes.

### CODE REVIEW

**Review by javerbukh — APPROVED** [review]
Code looks good and works well, thanks! I think I would just like to run the `get_spectra()` update by the team but otherwise this is good to merge.

File: docs/cubeviz/export_data.rst
Comment (pllim) [inline docs/cubeviz/export_data.rst:31]:
```suggestion
    subset1_spec1d = cubeviz.get_data(data_label=flux_data_label, 
```

**Review by pllim — COMMENTED** [review]
Seems reasonable.

File: jdaviz/configs/default/plugins/line_lists/line_lists.py
Comment (duytnguyendtn) [inline jdaviz/configs/default/plugins/line_lists/line_lists.py:169]: Ahh, I understand your desire to make `get_data` available via the app. Personally I think how it is here, while clunky, is probably a more robust way to implement it, but I could be convinced otherwise

Comment (rosteen) [reply jdaviz/configs/default/plugins/line_lists/line_lists.py:169]: So is that an approval or a change request? 😅

**Comment by rosteen** [comment]
@pllim I applied your suggestions, thanks.

**Review by duytnguyendtn — APPROVED** [review]
Code review looks good on my end. Basically ready to approve, but wondering if the 3.11 failure is a known one that I'm out of the loop on?

File: jdaviz/configs/default/plugins/line_lists/line_lists.py
Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/line_lists/line_lists.py:169]: Haha nothing to hold out on. Just a note

**Comment by rosteen** [comment]
> Code review looks good on my end. Basically ready to approve, but wondering if the 3.11 failure is a known one that I'm out of the loop on?

I hadn't noticed that error, but it appears to be unrelated. Looks like a unit order mismatch in Blackbody fitting.

**Comment by rosteen** [comment]
FYI I just updated the Cubeviz and Specviz example notebooks to accurately reflect the changes to `get_spectra`.

**Comment by pllim** [comment]
> Looks like a unit order mismatch in Blackbody fitting

Oh.... https://github.com/astropy/astropy/pull/14439 😬

**Comment by camipacifici** [comment]
The output from `get_spectra()` is much more intuitive than before!
I was a bit confused by the way masks are applied, but I understand now that the `True` stands for `masked`.
I would suggest to add a specific example to the docs on how to get out the specific portion of the spectrum that is highlighted as subset and say explicitly that that is where the mask is `False`.

**Comment by rosteen** [comment]
@camipacifici I added your suggestions to the Specviz exporting data section. If the additions sound good to you I'll go ahead and merge this.

File: docs/cubeviz/export_data.rst
Comment (bmorris3) [inline docs/cubeviz/export_data.rst:37]: I wasn't sure exactly what you meant here, so I'm tweaking it with my interpretation?
```suggestion
how to collapse the flux cube down to a one dimensional spectrum - this is not 
necessarily equivalent to the collapsed spectrum in the spectrum viewer, which 
may have used a different collapse statistic.
```

File: docs/specviz/export_data.rst
Comment (bmorris3) [inline docs/specviz/export_data.rst:48]: I think this makes sense for now. But since `Spectrum1D` already has a mask attribute, we could(/should?) make `specviz.load_spectrum` load the mask so that you don't have to also index by the not-mask. Does the parser do this already/is this step necessary?

**Review by bmorris3 — APPROVED** [review]
Looks good! Added one suggestion for now, and one comment that we can address after merge.

File: docs/specviz/export_data.rst
Comment (rosteen) [reply docs/specviz/export_data.rst:48]: The parser does load the mask (I believe you can show the mask and uncertainties, should be a toggle somewhere), but what Cami wanted to do was load in the data such that the are outside the subset is gone completely, which is what this does.

**Comment by camipacifici** [comment]
Looks good to me!

**Pull request merged**
